### PR TITLE
[SPARK-20719] [SQL] Support LIMIT ALL

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?
`LIMIT ALL` is the same as omitting the `LIMIT` clause. It is supported by both PrestgreSQL and Presto. This PR is to support it by adding it in the parser. 

### How was this patch tested?
Added a test case

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/inputs/limit.sql [hunk 1]
```diff
@@ -1,23 +1,27 @@
 
 -- limit on various data types
-select * from testdata limit 2;
-select * from arraydata limit 2;
-select * from mapdata limit 2;
+SELECT * FROM testdata LIMIT 2;
```
Comment (HyukjinKwon) [inline sql/core/src/test/resources/sql-tests/inputs/limit.sql]: I just wonder why these should be upper-cased just for curiosity. Is this way preferred?

Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/limit.sql]: Yes. This is preferred. All the SQL keywords are preferred to use the upper case.

Comment (HyukjinKwon) [inline sql/core/src/test/resources/sql-tests/inputs/limit.sql]: I see. Thank you.

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 2]
```diff
@@ -523,14 +523,6 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
...
   }
 
-  test("negative in LIMIT or TABLESAMPLE") {
-    val expected = "The limit expression must be equal to or greater than 0, but got -1"
-    var e = intercept[AnalysisException] {
-      sql("SELECT * FROM testData TABLESAMPLE (-1 rows)")
-    }.getMessage
-    assert(e.contains(expected))
-  }
-
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: I see. This was duplicated.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
+1, LGTM.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 3]
```diff
@@ -279,7 +279,12 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
 
     // LIMIT
     withWindow.optional(limit) {
-      Limit(typedVisit(limit), withWindow)
+      if (ALL != null) {
+        // LIMIT ALL is the same as omitting the LIMIT clause
```
Comment (hvanhovell) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: Can this be true? I don't think you need to check `ALL` if `limit != null`

Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: Sure. Thank you!

**Comment by hvanhovell** [comment]
LGTM - merging to master. Thanks!

**Pull request closed without merging**